Added example for secondary display using Presentation

jush opened the pull request:
<!--
Thanks for submitting a pull request!
Please fill out the sections below to complete your submission.
We appreciate your contributions!
-->

### Summary of changes

Added new example `SecondaryDisplayPresentationActivity` to showcase how to use [Presentation](https://developer.android.com/reference/android/app/Presentation) to show a `Mapview` on a secondary display.

The main difference is that `Presentation` inherits from `Dialog` and it's embedded in an `Activity` which doesn't inherit from `AppCompatActivity`. Therefore, lifecycle needs to be managed manually and also apply a theme that inherits from `Theme.AppCompat`.

### User impact (optional)

New demo:
![Mapview_on_Presentation](https://user-images.githubusercontent.com/691172/206663712-01e0f5ee-2bee-482e-948c-f817c97e63ee.png)


## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
    <!--
        | Before | After |
        | ----- | ----- |
        | <img src="" width = 250/> | <img src="" width = 250/> |
        or
        | <video src="" width = 250/> | <video src="" width = 250/> |
    -->
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [x] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

kiryldz commented on CHANGELOG.md:
```diff
@@ -3,6 +3,8 @@
 
 # main
+## Features ✨ and improvements 🏁
+* Added new example `SecondaryDisplayPresentationActivity` to showcase how to use [Presentation](https://developer.android.com/reference/android/app/Presentation) to show a `Mapview` on a secondary display. ([1905](https://github.com/mapbox/mapbox-maps-android/pull/1905))
```
sadly example are not part of the SDK so they do not deserve changelog entry...

yunikkk replied:
Why shouldn't we just add this functionality to the SDK though?

jush replied:
😢

jush replied:
@yunikkk do you mean that we should have our own Presentation that adds the right lifecycle events and expose that?

jush replied:
The issue I see here is that the `Presentation` is a class provided by Android itself.

Moreover, we have no control at all on the `Activity` that it is used to host that `Presentation` dialog and it doesn’t inherit from the `AppCompatActivity` one. It uses the standard/vanilla `Activity`.

I honestly don’t think that we should maintain a custom `Presentation`, I would rather see the users do it if they really need it ’cause they might need other customization in their `Presentation` implementation.

kiryldz commented on app/src/main/java/com/mapbox/maps/testapp/examples/SecondaryDisplayPresentationActivity.kt:
```diff
@@ -0,0 +1,201 @@
+import com.mapbox.maps.testapp.databinding.ActivitySecondaryDisplayPresentationBinding
+import com.mapbox.maps.testapp.databinding.ItemDisplayInfoBinding
+
+private const val TAG = "SecondaryDisplayPresent"
+
+private const val DISPLAY_INFO_TAG = "DisplayInfo"
```
guess we're mostly defining consts in companion object in the bottom of the file

kiryldz commented on app/src/main/java/com/mapbox/maps/testapp/examples/SecondaryDisplayPresentationActivity.kt:
```diff
@@ -0,0 +1,201 @@
+        presentation.show()
+        presentation
+      } catch (ex: WindowManager.InvalidDisplayException) {
+        Log.w(TAG, "Couldn't show presentation!  Display was removed in the meantime.", ex)
```
```suggestion
        Log.e(TAG, "Couldn't show presentation!  Display was removed in the meantime.", ex)
```

kiryldz commented on app/src/main/java/com/mapbox/maps/testapp/examples/SecondaryDisplayPresentationActivity.kt:
```diff
@@ -0,0 +1,201 @@
+  }
+  private val displayListener = object : DisplayManager.DisplayListener {
+    override fun onDisplayAdded(displayId: Int) {
+      Log.d(TAG, "Display #$displayId added.")
```
debug logs will not be shown in release - perhaps use info?
```suggestion
      Log.i(TAG, "Display #$displayId added.")
```

jush replied:
AFAIK, they're shown in release since we don't check if the tag is enabled or not.

kiryldz replied:
ah, you're using Android logger and not Common SDK one

jush replied:
Yes, I thought it would be better since this is an example that users might copy.

kiryldz commented on app/src/main/java/com/mapbox/maps/testapp/examples/SecondaryDisplayPresentationActivity.kt:
```diff
@@ -0,0 +1,201 @@
+  }
+
+  override fun onStop() {
+    super.onStop()
```
move super after dismissing presentation?

jush commented on app/src/main/java/com/mapbox/maps/testapp/examples/SecondaryDisplayPresentationActivity.kt:
```diff
@@ -0,0 +1,201 @@
+
+    override fun onStop() {
+      lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_STOP)
+      lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_DESTROY)
```
Yes, it's needed. I couldn't find a better place where to put the on destroy in a dialog...

paulrenn67 commented on app/src/main/java/com/mapbox/maps/testapp/examples/SecondaryDisplayPresentationActivity.kt:
```diff
@@ -0,0 +1,204 @@
+    } else {
+      secondaryDisplays.forEach { display ->
+        Log.d(TAG, "updatePresentation: Found display $display")
+        val displayInfoBinding =
```
could maybe use something like this to reduce repeated use of displayInfoBinding? Just my personal preference.
```
ItemDisplayInfoBinding.inflate(
            LayoutInflater.from(binding.root.context),
            binding.root,
            true
          ).apply {
             root.tag = DISPLAY_INFO_TAG
             displayInfo.text = "${display.name} [${display.displayId}]"
             displayBt.isEnabled = true
           // etc
```

kiryldz reviewed (approved):
Looks good as an example, couple of things I did oberve:
1. Perhaps add instructions to `SecondaryDisplayPresentationActivity` class (as javadoc) stating how to test it if there are no secondary displays (enable Simulate secondary display in Developer options)
2. When I did run an activity - I had `No secondary displays available`. When I minimized app and simulated one - it was displaying as expected but I still have `No secondary displays available` text, perhaps fix that

jush commented:
> Looks good as an example, couple of things I did observe:
> 
> 1. Perhaps add instructions to `SecondaryDisplayPresentationActivity` class (as javadoc) stating how to test it if there are no secondary displays (enable Simulate secondary display in Developer options)
> 2. When I did run an activity - I had `No secondary displays available`. When I minimized app and simulated one - it was displaying as expected but I still have `No secondary displays available` text, perhaps fix that

I have now fixed the issue 2 and added docs explaining couple of options to use this example.
